simulator-core/src/test/java/com/evbox/everon/ocpp/simulator/StationSimulatorFunctionalTest.java

```diff
@@ -685,4 +739,12 @@ private String toJson(Object object) {
     private void triggerUserAction(String stationId, UserMessage action) {
         stationSimulatorRunner.getStation(stationId).sendMessage(new StationMessage(stationId, StationMessage.Type.USER_ACTION, action));
     }
+
+    static class NotifyReportComparator implements Comparator<NotifyReportRequest>
```

Comment:
Do u still need this class?

Reply:
Yes, because this a functional test that verifies that all messages from 1 .. N are present, and the proper way to verify that is to sort all the message (or insert them into a set) and check for all values between 1 and N.

Yes, in our (weak) implementation , the messages will always arrive in order, but the test should be agnostic about that fact. Not to mention in the future if we switch to an async model, or add that random shuffling back, the test would fail